airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -183,16 +201,16 @@ public SourceDiscoverSchemaRead discoverSchemaForSourceFromSourceCreate(SourceCo
     final SourceConnection source = new SourceConnection()
         .withSourceDefinitionId(sourceCreate.getSourceDefinitionId())
         .withConfiguration(sourceCreate.getConnectionConfiguration());
-    final Job job = schedulerJobClient.createDiscoverSchemaJob(source, imageName);
-    return discoverJobToOutput(job);
+    final Pair<StandardDiscoverCatalogOutput, JobInfoRead> output =
+        executeSynchronousJob(() -> schedulerSynchronousJobClient.createDiscoverSchemaJob(source, imageName));
```

Comment:
can you do the transformation of the client output in the `xxxJobToOutput` instead of going through a pair? That would also look better if you stop using the exception to pass data. You will also be able to provide the metadata whether it is successful or not